src/hooks/useClientSideV3Trade.ts

```diff
@@ -57,16 +57,18 @@ export function useClientSideV3Trade<TTradeType extends TradeType>(
     gasRequired: chainId ? QUOTE_GAS_OVERRIDES[chainId] ?? DEFAULT_GAS_QUOTE : undefined,
   })
 
+  const currenciesAreTheSame = useMemo(
+    () => currencyIn && currencyOut && (currencyIn.equals(currencyOut) || currencyIn.wrapped.equals(currencyOut)),
```

Comment:
does `equals` always do the check we want here? will it checksum both sides?

Reply:
the currency entity checksums both sides when the asset is created. I think this might still let through a quote attempt if the same asset is provided as input and output, but only one is checksummed.

Reply:
note there is an optional argument on `Currency` creation that bypasses the checksum, but i think sdk-core should fix the address comparison to be case insensitive 🤔

Reply:
🤔 feels like we should also make it more explicit that this is for wraps/unwraps by making the var name `isWrap`. the semantics of calling it the same currency isnt totally accurate imo

and i know this was existing code but also doesnt the store already prevent the case where `currencyIn.equals(currencyOut)`?